### PR TITLE
Update nvq++ script to disallow nvqc submission from non-x86

### PR DESCRIPTION
**Pull request by bmhowe23**

### CODE REVIEW

File: tools/nvqpp/nvq++.in [hunk 1]
```diff
@@ -589,6 +589,13 @@ COMPILER_FLAGS="${CPPSTD} -Wno-c++20-extensions ${COMPILER_FLAGS}"
 # the resultant binary to target that specified backend.
 OBJS_TO_MERGE=""
 if [ -n "${TARGET_CONFIG}" ]; then
+	if [ "${TARGET_CONFIG}" == "nvqc" ]; then
+		LLVM_CONFIG="${LLVMBIN}llvm-config"
+		HOST_TARGET=$(${LLVM_CONFIG} --host-target)
+		if [ ${HOST_TARGET:0:6} != "x86_64" ]; then
+			error_exit "Cannot use nvqc target from non-x86_64 host at this time"
+		fi
+	fi
```
Comment (khalatepradnya) [inline tools/nvqpp/nvq++.in:598]: Q: Is there an issue tracking this? Perhaps we can add a comment linking with it so, its easier to find.

**Review by khalatepradnya — APPROVED** [review]
Looks good.

File: tools/nvqpp/nvq++.in [hunk 2]
```diff
@@ -589,6 +589,13 @@ COMPILER_FLAGS="${CPPSTD} -Wno-c++20-extensions ${COMPILER_FLAGS}"
 # the resultant binary to target that specified backend.
 OBJS_TO_MERGE=""
 if [ -n "${TARGET_CONFIG}" ]; then
+	if [ "${TARGET_CONFIG}" == "nvqc" ]; then
+		LLVM_CONFIG="${LLVMBIN}llvm-config"
+		HOST_TARGET=$(${LLVM_CONFIG} --host-target)
+		if [ ${HOST_TARGET:0:6} != "x86_64" ]; then
+			error_exit "Cannot use nvqc target from non-x86_64 host at this time"
```
Comment (khalatepradnya) [inline tools/nvqpp/nvq++.in:596]: [nit] I don't know the terminology used elsewhere with regards to `nvqc`, but, is `client` more suitable than `host`?

File: tools/nvqpp/nvq++.in (hunk 1 above)
Comment (bmhowe23) [reply tools/nvqpp/nvq++.in:600]: Good point. I wrote issue #1345 and referenced it in a new comment in 861fc1822.

File: tools/nvqpp/nvq++.in (hunk 2 above)
Comment (bmhowe23) [reply tools/nvqpp/nvq++.in]: Updated in 861fc1822

**Pull request merged**